src/lib/ledger_catchup/ledger_catchup.ml

```diff
@@ -98,14 +94,35 @@ module Make (Inputs : Inputs.S) :
             Deferred.return @@ Or_error.error_string message
         | Some queried_transitions ->
             let%bind queried_transitions_verified =
-              let staged_ledger =
-                Transition_frontier.best_tip frontier
-                |> Transition_frontier.Breadcrumb.staged_ledger
-              in
-              Deferred.Or_error.List.map queried_transitions
+              Deferred.Or_error.List.filter_map queried_transitions
                 ~f:(fun transition ->
-                  Transition_handler_validator.verify_transition ~staged_ledger
-                    ~transition )
+                  let open Deferred.Let_syntax in
+                  let transition_with_hash =
+                    With_hash.of_data transition
+                      ~hash_data:
+                        (Fn.compose Consensus.Mechanism.Protocol_state.hash
+                           External_transition.protocol_state)
+                  in
+                  match%map
+                    Transition_handler_validator.validate_transition ~logger
+                      ~frontier transition_with_hash
+                  with
+                  | `Valid ->
```

Comment:
shouldn't `Valid` return you a `External_transition.Verified.t`

Reply:
Could be modeled that way I guess.